src/components/Events/EventLevelMenu.vue

```diff
@@ -0,0 +1,71 @@
+<script lang="ts" setup>
+import {
+  EventLevelValue,
+  eventLevelValueMap,
+  eventLevels
+} from '/@/consts/eventLevel'
+
+interface Props {
+  eventLevel: EventLevelValue
+}
+
+defineProps<Props>()
+
+const emit = defineEmits<{
+  (e: 'update-public-status', value: EventLevelValue): void
+}>()
+</script>
+
+<template>
+  <div :class="$style.eventLevelMenu">
+    <div v-for="[key, value] in Object.entries(eventLevels)" :key="key">
+      <button
+        :class="$style.eventLevelMenuButton"
+        :disabled="eventLevel === key"
+        @click="emit('update-public-status', eventLevelValueMap[value.value])"
```

Comment:
`key` が `EventLevelValue` に相当するので、ここは単に `key` だけで大丈夫です :+1:
(`Object.entries` の型定義が弱くて key が string に推論されちゃうので、 `as EventLevelValue` が必要になっちゃうけど)
```suggestion
        @click="emit('update-public-status', key as EventLevelValue)"
```

一応
```ts
type Entries<T extends Record<PropertyKey, unknown>, Keys = keyof T> = Keys extends keyof T ? [Keys, T[Keys]][] : never
```
とかを定義して、`Object.entries(eventLevels) as Entries<typeof eventLevels>` とかやる方法もあるにはあるんですがそこまではしなくていいかな

Reply:
手っ取り早く治すために `as EventLevelValue` って書いちゃったんですが、厳密に (というか変更に強く) 書くなら `keyof typeof eventLevels` 使って
```suggestion
        @click="emit('update-public-status', key as keyof typeof eventLevels)"
```
って書いたほうがいいです 🤟